src/client/Microsoft.Identity.Client/Internal/Requests/ClientCredentialRequest.cs

```diff
@@ -143,13 +138,72 @@ private async Task<AuthenticationResult> GetAccessTokenAsync(
         {
             await ResolveAuthorityAsync().ConfigureAwait(false);
 
-            // Get a token from AAD
-            if (ServiceBundle.Config.AppTokenProvider == null)
+            AuthenticationResult authResult = null;
+            int retryCount = 0;
+
+            // Retry loop using the retry callback if configured
+            while (true)
```

Comment:
Should there be a hard limit on retries? just to be sure we dont get into an infinite loop?